syntax = "proto3";

import "common.proto";
import "interface.proto";

package Net;

// The LAG service definition.
service LagManagement {
    rpc CreateLag (Lag) returns (Result) {}
    rpc DeleteLag (Lag) returns (Result) {}

    rpc AddLagMember (LagMember) returns (Result) {}
    rpc RemoveLagMember (LagMember) returns (Result) {}
}

// The request message containing the LAG id.
message LagInstance {
    Iface iface = 1;
}

message Lag {
    string id = 1;
}

message LagMember {
    Lag lag = 1;
    repeated Iface members = 2;
}
